### PR TITLE
genpolicy: fix and re-enable create container UID verification

### PR DESCRIPTION
**Pull request by danmihai1**
1. Add ability to pull UID:GID values from /etc/passwd, from container images.
2. Add genpolicy test for a container image that specifies a user name that must be converted to a UID value.
3. Re-enable CreateContainerRequest UID verification, now that the behavior from #9928 has been addressed by this PR.

Most of these changes have been cherry-picked from #9955 - thanks @HernanGatta !

Also, thanks to @burgerdev for #9928, for reviewing these changes, and for the additional ideas for possible future improvements tracked by #10290 !

### CODE REVIEW

**Review by burgerdev — APPROVED** [review]
LGTM - I can take a stab at the points in #10290 after this is merged.

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod-spark.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+apiVersion: v1
+kind: Pod
+metadata:
+  name: policy-pod-spark
+spec:
+  terminationGracePeriodSeconds: 0
+  runtimeClassName: kata
+  containers:
+    - name: spark
+      image: docker.io/spark:latest
```
[the remark this answers is not in the document]
Comment (ChengyuZhu6) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod-spark.yaml:15]: I agree with the concerns. Could you clarify why we're using the spark image specifically?

File: src/tools/genpolicy/src/registry.rs [hunk 2]
```diff
@@ -525,3 +641,31 @@ fn build_auth(reference: &Reference) -> RegistryAuth {
...
+                "Incorrect passwd record, expected 7 fields, got {}",
+                field_count
+            ));
+        }
+
+        records.push(PasswdRecord {
+            user: fields[0].to_string(),
+            validate: fields[1] == "x",
+            uid: fields[2].parse().unwrap(),
+            gid: fields[3].parse().unwrap(),
```
[the remark this answers is not in the document]
Comment (danmihai1) [reply src/tools/genpolicy/src/registry.rs:663]: I see your point now - thanks for clarifying! I have just added this into https://github.com/kata-containers/kata-containers/issues/10290, for future improvements.

File: tests/integration/kubernetes/k8s-policy-pod-docker-io.bats [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	pod_name="policy-pod-spark"
+
+	get_pod_config_dir
+	policy_settings_dir="$(create_tmp_policy_settings_dir "${pod_config_dir}")"
+	
+	add_requests_to_policy_settings "${policy_settings_dir}" "ReadStreamRequest"
+
+	correct_pod_yaml="${pod_config_dir}/k8s-policy-pod-spark.yaml"
+    auto_generate_policy "${policy_settings_dir}" "${correct_pod_yaml}" "${correct_configmap_yaml}"
```
[the remark this answers is not in the document]
Comment (danmihai1) [reply tests/integration/kubernetes/k8s-policy-pod-docker-io.bats]: @burgerdev , I replaced this entire test file by using your idea of using the prometheus image - thanks!

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod-spark.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danmihai1) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-pod-spark.yaml]: I wasn't aware of the prometheus image using a username too. I agree that image works better for CI testing - thank you!

**Comment by danmihai1** [comment]
> LGTM - I can take a stab at the points in #10290 after this is merged.

@burgerdev , thanks for all the help!

**Review by Redent0r — APPROVED** [review]

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM thanks @danmihai1

**Comment by danmihai1** [comment]
@ChengyuZhu6 , @burgerdev , at a quick look quay.io/prometheus/busybox doesn't seem to work on the CI TEEs. We get on all 3 of them:

```
2024-09-12T00:34:45.0213879Z # Events:
2024-09-12T00:34:45.0214305Z #   Type     Reason     Age                From               Message
2024-09-12T00:34:45.0215011Z #   ----     ------     ----               ----               -------
2024-09-12T00:34:45.0216196Z #   Normal   Scheduled  90s                default-scheduler  Successfully assigned kata-containers-k8s-tests/policy-pod to kata-ci-amd-milan-001
2024-09-12T00:34:45.0217594Z #   Normal   Pulled     78s                kubelet            Successfully pulled image "quay.io/prometheus/prometheus:latest" in 977.86791ms
2024-09-12T00:34:45.0220056Z #   Warning  Failed     78s                kubelet            Error: failed to create containerd container: mount callback failed on /var/lib/containerd/tmpmounts/containerd-mount175908297: open /var/lib/containerd/tmpmounts/containerd-mount175908297/etc/passwd: no such file or directory
```

I will run the tests on https://github.com/kata-containers/kata-containers/pull/10296 to verify that theory.

**Comment by ChengyuZhu6** [comment]
There is an issue in the CI.

**Review by GabyCT — APPROVED** [review]

**Pull request merged**